src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,119 @@
+                                + "event <desc> /from <start> /to <end>");
+                    }
+                    String[] thirdSplit = secondSplit[1].split(" +/to +");
+                    if (thirdSplit.length != 2) {
+                        throw new ParserException("Incorrect syntax!\n"
+                                + "Please use this syntax:\n"
+                                + "event <desc> /from <start> /to <end>");
+                    }
+                    try {
+                        LocalDateTime start = LocalDateTime.parse(thirdSplit[0], DukeEnvironmentConstants.FORMATTER1);
+                        LocalDateTime end = LocalDateTime.parse(thirdSplit[1], DukeEnvironmentConstants.FORMATTER1);
+                        if (end.isBefore(start)) {
+                            throw new ParserException(INVALID_DATE_ORDER);
+                        }
+                    } catch (DateTimeParseException e) {
+                        throw new ParserException(INVALID_DATE_FORMAT);
+                    }
+                    return new AddToListCommand(DukeEnvironmentConstants.TaskType.EVENT,
+                            new String[]{secondSplit[0], thirdSplit[0], thirdSplit[1]});
+                }
+            } else if (splitInput[0].equals("deadline")) {
+                if (splitInput.length < 2) {
+                    throw new ParserException(EMPTY_DEADLINE_DESC);
+                } else {
+                    String[] secondSplit = splitInput[1].split(" +/by +", 2);
+                    if (secondSplit.length != 2) {
+                        throw new ParserException("Incorrect syntax!\n"
+                                + "Please use this syntax:\n"
+                                + "deadline <desc> /by <deadline>");
+                    }
+                    try {
+                        LocalDateTime.parse(secondSplit[1], DukeEnvironmentConstants.FORMATTER1);
+                    } catch (DateTimeParseException e) {
+                        throw new ParserException(INVALID_DATE_FORMAT);
+                    }
+                    return new AddToListCommand(DukeEnvironmentConstants.TaskType.DEADLINE,
+                            new String[]{secondSplit[0], secondSplit[1]});
+                }
+            } else if (splitInput[0].equals("todo")) {
+                if (splitInput.length < 2) {
+                    throw new ParserException(EMPTY_TODO_DESC);
+                } else {
+                    return new AddToListCommand(DukeEnvironmentConstants.TaskType.TODO, new String[]{splitInput[1]});
+                }
+            } else if (splitInput[0].equals("find")) {
+                if (splitInput.length < 2) {
+                    throw new ParserException(EMPTY_FIND_QUERY);
+                } else {
+                    return new FindCommand(splitInput[1]);
+                }
+            } else {
+                throw new ParserException(NO_VALID_COMMAND);
+            }
+        }
+    }
```

Comment:
Perhaps can consider splitting this method into separate methods for parsing each type of command? It could help reduce the length, nesting and improve readability